src/hyperlight_host/src/sandbox/initialized_multi_use.rs

```diff
@@ -512,10 +601,192 @@ impl MultiUseSandbox {
+            // - Data structures partially updated (e.g., hash table with inconsistent entry count)
+            // - Resources unreleased (file handles, mutexes, network connections)
+            // - Global/static variables in undefined state
+            //
+            // ExecutionCanceledByHost: Guest execution was interrupted mid-operation, potentially leaving:
+            // - Memory allocations that were in progress of being allocated/freed
+            // - I/O operations in inconsistent state (partial writes, half-open connections)
+            // - Critical sections or mutexes that were being acquired/released
+            // - Memory allocator metadata in corrupted state
+            HyperlightError::GuestAborted(_, _) | HyperlightError::ExecutionCanceledByHost() => {
+                true
+            }
+
+            // All other errors do not poison the sandbox
+            HyperlightError::AnyhowError(_)
+            | HyperlightError::BoundsCheckFailed(_, _)
+            | HyperlightError::CheckedAddOverflow(_, _)
+            | HyperlightError::CStringConversionError(_)
+            | HyperlightError::Error(_)
+            | HyperlightError::ExecutionAccessViolation(_)
+            | HyperlightError::FailedToGetValueFromParameter()
+            | HyperlightError::FieldIsMissingInGuestLogData(_)
+            | HyperlightError::GuestError(_, _)
+            | HyperlightError::GuestExecutionHungOnHostFunctionCall()
+            | HyperlightError::GuestFunctionCallAlreadyInProgress()
+            | HyperlightError::GuestInterfaceUnsupportedType(_)
+            | HyperlightError::GuestOffsetIsInvalid(_)
+            | HyperlightError::HostFunctionNotFound(_)
+            | HyperlightError::IOError(_)
+            | HyperlightError::IntConversionFailure(_)
+            | HyperlightError::InvalidFlatBuffer(_)
+            | HyperlightError::JsonConversionFailure(_)
+            | HyperlightError::LockAttemptFailed(_)
+            | HyperlightError::MemoryAccessViolation(_, _, _)
+            | HyperlightError::MemoryAllocationFailed(_)
+            | HyperlightError::MemoryProtectionFailed(_)
+            | HyperlightError::MemoryRequestTooBig(_, _)
+            | HyperlightError::MetricNotFound(_)
+            | HyperlightError::MmapFailed(_)
+            | HyperlightError::MprotectFailed(_)
+            | HyperlightError::NoHypervisorFound()
+            | HyperlightError::NoMemorySnapshot
+            | HyperlightError::ParameterValueConversionFailure(_, _)
+            | HyperlightError::PEFileProcessingFailure(_)
+            | HyperlightError::PoisonedSandbox
+            | HyperlightError::RawPointerLessThanBaseAddress(_, _)
+            | HyperlightError::RefCellBorrowFailed(_)
+            | HyperlightError::RefCellMutBorrowFailed(_)
+            | HyperlightError::ReturnValueConversionFailure(_, _)
+            | HyperlightError::StackOverflow()
+            | HyperlightError::SnapshotSandboxMismatch
```

Comment:
Some of these like this one seem like a data corruption. I am a little unclear on now we determine if it puts a function into a "posioned" state.